Assets/SEE/GraphProviders/VCSGraphProvider.cs

```diff
@@ -0,0 +1,421 @@
+                return Enumerable.Empty<SEEToken>();
+            }
+        }
+
+        /// <summary>
+        /// Adds Halstead, McCabe and lines of code metrics to the corresponding node for the supported TokenLanguages in <paramref name="graph"/>.
+        /// Otherwise, metrics are not available.
+        /// </summary>
+        /// <param name="graph">The graph where the metric should be added.</param>
+        /// <param name="repository">The repository from which the file content is retrieved.</param>
+        /// <param name="commitID">The commitID where the files exist.</param>
+        protected static void AddMetricsToNode(Graph graph, Repository repository, string commitID)
+        {
+            HalsteadMetrics halsteadMetrics;
+            foreach (Node node in graph.Nodes())
+            {
+                if (node.Type == "file")
+                {
+                    string filePath = node.ID.Replace('\\', '/');
+                    IEnumerable<SEEToken> tokens;
+                    TokenLanguage language;
+                    try
+                    {
+                        language = TokenLanguage.FromFileExtension(Path.GetExtension(filePath).TrimStart('.'), true);
+                        tokens = RetrieveTokens(filePath, repository, commitID, language);
+                        int complexity = CalculateMcCabeComplexity(tokens);
+                        int linesOfCode = CalculateLinesOfCode(tokens);
+                        halsteadMetrics = CalculateHalsteadMetrics(tokens);
+                        node.SetInt("Lines of Code", linesOfCode);
+                        node.SetInt("McCabe Complexity", complexity);
+                        node.SetInt("Halstead Distinct Operators", halsteadMetrics.DistinctOperators);
+                        node.SetInt("Halstead Distinct Operands", halsteadMetrics.DistinctOperands);
+                        node.SetInt("Halstead Total Operators", halsteadMetrics.TotalOperators);
+                        node.SetInt("Halstead Total Operands", halsteadMetrics.TotalOperands);
+                        node.SetInt("Halstead Program Vocabulary", halsteadMetrics.ProgramVocabulary);
+                        node.SetInt("Halstead Program Length", halsteadMetrics.ProgramLength);
+                        node.SetFloat("Halstead Calculated Estimated Program Length", halsteadMetrics.EstimatedProgramLength);
+                        node.SetFloat("Halstead Volume", halsteadMetrics.Volume);
+                        node.SetFloat("Halstead Difficulty", halsteadMetrics.Difficulty);
+                        node.SetFloat("Halstead Effort", halsteadMetrics.Effort);
+                        node.SetFloat("Halstead Time Required to Program", halsteadMetrics.TimeRequiredToProgram);
+                        node.SetFloat("Halstead Number of Delivered Bugs", halsteadMetrics.NumberOfDeliveredBugs);
+                    }
+                    catch
+                    {
+
+                    }
```

Comment:
That is much better, however, an even simpler solution IMO for this would be to replace the `true` parameter with `false` (since this is the default, it can also be omitted entirely) in the `FromFileExtension` call. You can then check if the language is `Plain` and, if so, continue with the next path.
It could look like this:
```csharp
TokenLanguage language = TokenLanguage.FromFileExtension(Path.GetExtension(filePath).TrimStart('.'), false);
if (language == TokenLanguage.Plain)
{
    continue;
}
tokens = RetrieveTokens(filePath, repository, commitID, language);
int complexity = CalculateMcCabeComplexity(tokens);
int linesOfCode = CalculateLinesOfCode(tokens);
// [...] Rest of code here
```